### PR TITLE
Update 1/3 & 2/3 for Milestone Starnuti 2016

### LINKED ISSUE #27: Show values on bar-settings in the scenario editor
In #3 , it is possible to set various properties of the "game" in GUI, make a textfield showing these values as they are set

### LINKED ISSUE #36: Set default values for the distractors in Distraction Task
These should be "optimal" values that will be used in reproducible experiments. 
- [x] "Default" button
- [ ] apply the defaults on our experiment by default
- [ ] optimize the default parameters

### PR DESCRIPTION
**Pull request by breznak**
Menu-related items. 

Fixes #27 
Partially #32 (not working here) 
Fixes #36 
Partially #22 
Implements code for #23 (just will need to make this more usable)

### CODE REVIEW

File: assets/Interface/DistractionScreenGUI.xml [hunk 1]
```diff
@@ -7,11 +7,17 @@
...
                 <panel id="weather_panel" childLayout="absolute" backgroundColor="#333333ff" width="177px" x="13" y="66" style="nifty-panel" height="334px">
                     <control name="horizontalSlider" id="Slider_fog" width="150px" x="3" y="290"/>
-                    <control name="checkbox" id="CheckBox_fog" x="55" y="256"/>
+                    <control name="checkbox" id="CheckBox_fog" x="37" y="256"/>
+					<control name="listBox" id="Value_fog" width="35px" x="117" y="256" height="25px" vertical="optional" horizontal="optional" displayItems="1" selection="Disabled"/>
+					
                     <control name="horizontalSlider" id="Slider_snow" width="150px" x="3" y="205"/>
-                    <control name="checkbox" id="CheckBox_snow" x="55" y="172"/>
+                    <control name="checkbox" id="CheckBox_snow" x="37" y="172"/>
+					<control name="listBox" id="Value_snow" width="35px" x="117" y="172" height="25px" vertical="optional" horizontal="optional" displayItems="1" selection="Disabled"/>
```
Comment (breznak) [inline assets/Interface/DistractionScreenGUI.xml:15]: this is to show the value of the bar-slider?

File: assets/Interface/DistractionScreenGUI.xml [hunk 2]
```diff
@@ -28,34 +34,48 @@
                 <control name="label" id="GLabel19" width="60px" x="11" y="333" text="Fog" height="25px" font="aurulent-sans-16.fnt"/>
                 <control name="label" id="GLabel17" width="60px" x="15" y="248" text="Snow" height="25px" font="aurulent-sans-16.fnt"/>
                 <control name="label" id="GLabel18" width="60px" x="15" y="158" text="Rain" height="25px" font="aurulent-sans-16.fnt"/>
+				
                 <control name="button" id="StartButton" childLayout="center" x="529" width="120px" y="560" label="Start" height="31px">
 					<interact onClick="clickStartButton()" />
 				</control>
+				<control name="button" id="InstructionButton" childLayout="center" x="529" width="120px" y="480" label="Instruction" height="31px">
+					<interact onClick="clickInstButton()" />
```
Comment (breznak) [inline assets/Interface/DistractionScreenGUI.xml:42]: this will show the "Instruction screen", right? #32 

For the "instructions" and "Default values", where are the values/text read from? I'd like to have a text/xml file where this is written...

File: src/cz/cvut/cognitive/override/DistractionScreenGUIController.java [hunk 3]
```diff
@@ -365,6 +439,75 @@ public void clickStartButton()
...
+                
+                CheckBox_rain.check();
+                Slider_rain.setValue(20);
+                DistractionSettings.setIntensityRain(20);
+                DistractionSettings.setRain(true);
+                
+                CheckBox_box.check();
+                Slider_box.setValue(30);
+                DistractionSettings.setProbabilityBox(20);
+                DistractionSettings.setBox(true);
```
Comment (breznak) [inline src/cz/cvut/cognitive/override/DistractionScreenGUIController.java:456]: why are these here, and not in the "resetOptions()"?

**Review by breznak — COMMENTED** [review]
Thanks! I've resoved the merge conflict and reviewed the PR. 

Did you make any changes to the README.txt? github doesn't show any. I'm changing it to the .md format

**Comment by breznak** [comment]
So review for the "Default options" #36 : 
- see the updated issue please, those need to be completed (although the import from XML would be nice, it is not necessary, #45 )

**Comment by breznak** [comment]
Review for "Instructions" #32 
- it does not work for me (does nothing), there is no action `clickInstButton()`

File: assets/Interface/DistractionScreenGUI.xml (hunk 1 above)
Comment (JohnnyMarek) [reply assets/Interface/DistractionScreenGUI.xml:15]: Line 15, listbox: shows the value on the slider next to (above) it.

File: assets/Interface/DistractionScreenGUI.xml (hunk 2 above)
Comment (JohnnyMarek) [reply assets/Interface/DistractionScreenGUI.xml:42]: Yes, first i create control for the button and then add interact function to it which is implemented in java. Default values are read from the code, will make it to be from txt.

File: src/cz/cvut/cognitive/override/DistractionScreenGUIController.java (hunk 3 above)
Comment (JohnnyMarek) [reply src/cz/cvut/cognitive/override/DistractionScreenGUIController.java:456]: This is how i assign the default values. first i reset all options and then set them to these values. Will try to work it from txt file

File: assets/Interface/DistractionScreenGUI.xml (hunk 2 above)
Comment (breznak) [reply assets/Interface/DistractionScreenGUI.xml:42]: @JohnnyMarek please add the "read from TXT" for Instructions & default values

File: src/cz/cvut/cognitive/distractors/BoxDistraction.java [hunk 4]
```diff
@@ -99,9 +99,9 @@ public BoxDistraction(Simulator sim, float x, float y, float z, float mass, Stri
      * despawn in earlier call).
      */
     @Override
-    public void update (float tpf, float probability){
+    public void update (float tpf){
         int n = (int)(Math.random() * 100) + 1;
-            if (n <= probability){ 
+            if (n <= 0.5f){ //FIXME
```
Comment (breznak) [inline src/cz/cvut/cognitive/distractors/BoxDistraction.java]: There was a merge error above, which I quickly fixed here. Please add correct actual value here.

File: src/cz/cvut/cognitive/distractors/CollectObjectDistraction.java [hunk 5]
```diff
@@ -56,21 +60,33 @@
...
         this.car = sim.getCar();
         this.manager = sim.getAssetManager();
         this.camera = sim.getCamera();
-
+        
+        rewardSoundNode = new AudioNode(manager,"Sounds/FF_Victory.wav",false);
+        rewardSoundNode.setLooping(false);
+        rewardSoundNode.setPositional(false);
+        
+        rewardText = new BitmapText(manager.loadFont("Interface/Fonts/Default.fnt"), false);
```
Comment (breznak) [inline src/cz/cvut/cognitive/distractors/CollectObjectDistraction.java]: FIXME here and elsewhere: please use `path"+File.separator+"next` instead of `/` , it caused us some problems on other platforms (WIn?)

File: src/cz/cvut/cognitive/distractors/DistractionClass.java [hunk 6]
```diff
@@ -13,7 +13,26 @@
...
+
+    public float getDisProbability() {
+        return disProbability;
+    }
+
+    public void setDisProbability(float disProbability) {
+        this.disProbability = disProbability;
+    }
+   
+    public abstract void update(float tpf);
```
Comment (breznak) [inline src/cz/cvut/cognitive/distractors/DistractionClass.java]: Let's not worry too much about DistractionClass, I have many changes in #56

File: src/cz/cvut/cognitive/distractors/ListOfDistractions.java [hunk 7]
```diff
@@ -38,73 +38,66 @@ public void initialize(){
...
+        int k = 0;
+        for (int i = 0; i < currentIndex; i++) {
+            int n = (int)(Math.random() * 100) + 1;
+            System.out.println(distractors[i].getDisProbability()+ " vs " + n);
+            if (distractors[i].getDisProbability() > n) dist_index[k++]=i;
+            
+        }
+        if (k != 0){
+            int n = (int)(Math.random() * k) + 1;
+            distractors[dist_index[n]].update(tpf);
```
Comment (breznak) [inline src/cz/cvut/cognitive/distractors/ListOfDistractions.java]: We are duplicit here, I'll merge this and #56 later.

**Review by breznak — COMMENTED** [review]
The info screen is missing, 
otherwise its :+1:

**Comment by breznak** [comment]
@JohnnyMarek nice work with the rewards! :tada: 

We have some overlap with #56 , please fix just the review comments, I'll merge this, then #47 and then #56. After that we weill focus on fixing remaining issues.

**Comment by breznak** [comment]
@JohnnyMarek do you have any pending changes for this PR? Otherwise I'll make the fixes and merge it. 
- [x] use File.separator
- [x] set default values by default in our task

**Comment by JohnnyMarek** [comment]
please merge what you have and let me know what to download. I will apply all changes to that branch and send it with these fixes

**Comment by breznak** [comment]
...I don;t have anythong now, I mean if I should make the fixes (in case you dont have it already)
